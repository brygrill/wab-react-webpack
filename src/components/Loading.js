import React from 'react';
import PropTypes from 'prop-types';
import { Dimmer, Loader } from 'semantic-ui-react';

const Loading = props => {
  return (
    <Dimmer active inverted>
      <Loader size="small">{props.content}</Loader>
    </Dimmer>
  );
};

Loading.propTypes = {
  content: PropTypes.string,
};

Loading.defaultProps = {
  content: 'Loading',
};

export default Loading;
