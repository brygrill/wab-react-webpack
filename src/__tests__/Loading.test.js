import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Loading from '../components/Loading';

describe('<Loading />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Loading />, div);
  });

  it('shows spinning loader', () => {
    const tree = renderer.create(<Loading />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
