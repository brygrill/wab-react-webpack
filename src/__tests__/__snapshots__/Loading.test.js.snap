// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> shows spinning loader 1`] = `
<div
  className="ui active transition visible inverted dimmer"
  onClick={[Function]}
>
  <div
    className="content"
  >
    <div
      className="center"
    >
      <div
        className="ui small text loader"
      >
        Loading
      </div>
    </div>
  </div>
</div>
`;
